src/weight_windows.cpp

```diff
@@ -624,20 +625,45 @@ void WeightWindows::update_magic(
   auto mesh_vols = this->mesh()->volumes();
 
   int e_bins = new_bounds.shape()[0];
-  for (int e = 0; e < e_bins; e++) {
-    // select all
-    auto group_view = xt::view(new_bounds, e);
 
-    // divide by volume of mesh elements
-    for (int i = 0; i < group_view.size(); i++) {
-      group_view[i] /= mesh_vols[i];
+  if (settings::solver_type == SolverType::MONTE_CARLO ||
+      !FlatSourceDomain::adjoint_) {
```

Comment:
Would it make more sense to pass in `method_` so that this could just be a check on the method rather than determining what should be used in a roundabout manner?

Reply:
I had originally wanted to do that, but realized that `method_` is a member of the `WeightWindowGenerator` class, whereas this function is part of the `WeightWindow` class so doesn't directly have access to the `method_` field. In theory we can also make the `method_` a member of the `WeightWindow` class as well, but this would involve changing the API I think as `WeightWindow` objects need to be able to be made from the CAPI. I had similarly considered that we could have the caller pass in the method as an argument to the `update_magic()` function, but again this would change the CAPI as that method is exposed. Given that there was an easy way of deducing the type of weight window method to apply given other global settings, it seemed a lesser evil to avoid changing the API.

Reply:
Given what you pointed out (`update_magic` can be changed without affecting the C API), can we go with the approach of passing in `method` as an argument to `update_magic` (or whatever its new name will be)?

Reply:
I've refactored the `update_magic` function to now be `update_weights`. The new function also includes an argument for passing in the method type, with the internal logic being based off the argument. The CAPI wrapper is still called `openmc_weight_windows_update_magic()`, but I think that's ok. Generating ww's with MAGIC via the CAPI seems like a good workflow, as you can visualize the weights and fluxes after each MC solve to see how they're progressing and decide how many iterations you need. There isn't that sort of use case for FW-CADIS, as there's no need to iterate over multiple solves. As such, only MAGIC would typically be called via the CAPI, so it's ok to me if there is no way of doing FW-CADIS via CAPI.